crates/core/component/funding/src/component/liquidity_tournament/rewards/mod.rs

```diff
@@ -1,64 +1,118 @@
 use anyhow::anyhow;
 use cnidarium::{StateRead, StateWrite};
 use futures::stream::StreamExt as _;
+use futures::{Stream, TryStreamExt};
 use penumbra_sdk_asset::{asset, Value, STAKING_TOKEN_ASSET_ID};
 use penumbra_sdk_community_pool::component::StateWriteExt as _;
 use penumbra_sdk_dex::{component::LqtRead as _, lp::position};
 use penumbra_sdk_distributions::component::StateReadExt as _;
 use penumbra_sdk_keys::Address;
+use penumbra_sdk_num::fixpoint::U128x128;
 use penumbra_sdk_sct::component::clock::EpochRead as _;
 use penumbra_sdk_txhash::TransactionId;
 
 use super::bank::Bank as _;
 use super::votes::StateReadExt as _;
 use crate::component::StateReadExt as _;
+use crate::params::LiquidityTournamentParameters;
 
-mod gauge;
-use gauge::{Gauge, Share};
+fn create_share(portion: impl Into<U128x128>, total: impl Into<U128x128>) -> U128x128 {
+    U128x128::ratio(portion.into(), total.into().max(U128x128::from(1u64)))
+        .expect("max(x, 1) cannot be 0")
+}
 
-async fn position_shares(
+async fn relevant_votes_for_asset(
     state: impl StateRead,
+    params: &LiquidityTournamentParameters,
     epoch: u64,
     asset: asset::Id,
-    top_n: usize,
-) -> anyhow::Result<Vec<(Share, position::Id)>> {
-    let mut stream = state.positions_by_volume_stream(epoch, asset)?.take(top_n);
-    let mut total = 0u128;
-    // This will end up containing the volume of each lp.
-    let mut tallies = Vec::new();
-    while let Some(x) = stream.next().await {
-        let (_, lp, volume) = x?;
-        let volume = volume.value();
-        total += volume;
-        tallies.push((Share::from(volume), lp));
+) -> anyhow::Result<u128> {
```

Comment:
I think we should use our `Amount` type everywhere, mostly a lesson learned from doing the 64 -> 128 bits migration